pkgs/stdenv/generic/check-meta.nix

```diff
@@ -120,12 +120,19 @@ let
     '';
 
   remediate_allowlist = allow_attr: rebuild_amendment: attrs:
+    let
+      flakeNote = "
+   Note: For `nix shell`, `nix build`, `nix develop` or any other Nix 2.4+
+   (Flake) command, `--impure` must be passed in order to read this
+   environment variable.
+      ";
+    in
     ''
       a) To temporarily allow ${remediation_phrase allow_attr}, you can use an environment variable
          for a single invocation of the nix tools.
 
            $ export ${remediation_env_var allow_attr}=1
-
+           ${lib.optionalString (! builtins ? currentSystem) flakeNote}
```

Comment:
I don't see an issue of using this "hack", but maybe it would be better to add a comment saying that maybe we should substitute this later for `isFlake` function.

Reply:
How do you add a comment into a multiline string? I would be happy to do that if you could figure this out.

Reply:
Maybe we could convert this to a function inside a let and put the comment inside the function.

Reply:
Anyway, almost sure this works too:

```
${# this is a comment
  lib.optionalString (! builtins ? currentSystem) flakeNote}
```

Reply:
I have tidied the code up in the latest force push in a way that should satisfy you. I have now put the optional string into the `let` block for `flakeNote`, which is also where I have added the comment. Now all I do to print it is `${flakeNote}` which keeps the formatting as sensible as it can be.